resources/default_pipeline_environment.yml

```diff
@@ -110,63 +110,65 @@ steps:
       - 'tests'
     testPackage: 'piper-bats'
   checksPublishResults:
-    aggregation:
-      active: true
-      thresholds:
-        fail:
-          high: '0'
     tasks:
       pattern: '**/*.java'
       low: ''
       normal: 'TODO,REVISE,XXX'
       high: 'FIXME'
       archive: true
       active: false
-      thresholds:
-        fail:
-          high: '0'
+      qualityGates:
+        - threshold: 1
+          type: 'TOTAL_HIGH'
+          unstable: false
     pmd:
       pattern: '**/target/pmd.xml'
       archive: true
       active: false
-      thresholds:
-        fail:
-          high: '0'
+      qualityGates:
+        - threshold: 1
+          type: 'TOTAL_HIGH'
+          unstable: false
     cpd:
       pattern: '**/target/cpd.xml'
       archive: true
       active: false
-      thresholds:
-        fail:
-          high: '0'
+      qualityGates:
+        - threshold: 1
+          type: 'TOTAL_HIGH'
+          unstable: false
     findbugs:
       pattern: '**/target/findbugsXml.xml, **/target/findbugs.xml'
       archive: true
       active: false
-      thresholds:
-        fail:
-          high: '0'
+      qualityGates:
+        - threshold: 1
+          type: 'TOTAL_HIGH'
+          unstable: false
     checkstyle:
       pattern: '**/target/checkstyle-result.xml'
       archive: true
       active: false
-      thresholds:
-        fail:
-          high: '0'
+      qualityGates:
+        - threshold: 1
+          type: 'TOTAL_ERROR'
```

Comment:
Out of curiosity: Why here TOTAL_ERROR and not TOTAL_HIGH as everywhere eles?

Reply:
I changed to ERROR as all findings in my test run where categorised as `ERROR` and not `HIGH`. I'm also unsure what is best here..

Reply:
So Checkstyle uses a severity of `ERROR`, `WARNING` and `INFO` which is mapped to `ERROR`, `WARNING_NORMAL` and `WARNING_LOW` by this code: 

https://github.com/jenkinsci/analysis-model/blob/530da69cff6c17b8a5a334f71c1208efeb55e7c3/src/main/java/edu/hm/hafner/analysis/Severity.java#L95-L106

Reply:
I added q-gates for `TOTAL_ERROR` and `TOTAL_HIGH` on all tools.